Sources/MapboxMaps/Foundation/MapView+Snapshot.swift

```diff
@@ -0,0 +1,63 @@
+@_implementationOnly import MapboxCommon_Private
+
+@available(iOSApplicationExtension, unavailable)
+extension MapView {
+
+    /// Errors related to rendered snapshots
+    @_spi(Experimental) public enum RenderedSnapshotError: Error {
+        /// No metal view available. Catastrophic error.
+        case noMetalView
+
+        /// Metal texture not present in mapView.
+        case invalidTexture
+
+        /// Texture failed to convert to CGImage
+        case textureConversionFailed
+
+        /// Metal validation is enabled, unsupported configuration
+        case metalValidationEnabled
+
+        /// Converted image is empty in snapshot
+        case convertedImageIsEmpty
+    }
+
+    /// Synchronously captures the last rendered map view (if available) and constructs a `UIImage` if successful.
+    /// - NOTE: This API must be called on main thread
+    @_spi(Experimental) public func snapshot() -> Result<UIImage, RenderedSnapshotError> {
```

Comment:
Rather than returning Result synchronously, can we make this a throwing method? We made the same choice several months ago during the public API cleanup.

Reply:
Sure! Will update.

Reply:
Addressed